File: sdk/samples/WFSamples/Technologies/Activities/Compensation/VB/Compensation/PurchaseOrder.vb
'----------------------------------------------------------------------
'   This file is part of the Windows Workflow Foundation SDK Code Samples.
'  
'   Copyright (C) Microsoft Corporation.  All rights reserved.
'  
' This source code is intended only as a supplement to Microsoft
' Development Tools and/or on-line documentation.  See these other
' materials for detailed information regarding Microsoft code samples.
'  
' THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
' KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
' IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
' PARTICULAR PURPOSE.
'----------------------------------------------------------------------

Imports System
Imports System.Runtime.Serialization
Imports System.Workflow.Activities
Namespace Microsoft.Samples.Workflow.Compensation
    Public Class PurchaseOrder
        Inherits SequentialWorkflowActivity

        Private Sub ReceiveOrderHandler(ByVal sender As System.Object, ByVal e As System.EventArgs)
            Console.WriteLine("Received order")
        End Sub

        Private Sub WithdrawFundsHandler(ByVal sender As System.Object, ByVal e As System.EventArgs)
            Console.WriteLine("Withdraw funds")
        End Sub

        Private Sub ProductCheckHandler(ByVal sender As System.Object, ByVal e As System.Workflow.Activities.ConditionalEventArgs)
            e.Result = False
        End Sub

        Private Sub ShipHandler(ByVal sender As System.Object, ByVal e As System.EventArgs)
            Console.WriteLine("Ship order")
        End Sub


        Private Sub DiscontinuedProduct_ExecuteCode(ByVal sender As System.Object, ByVal e As System.EventArgs)
            Console.WriteLine("Product discontinued")
        End Sub

        Private Sub RefundHandler(ByVal sender As System.Object, ByVal e As System.EventArgs)
            Console.WriteLine("Refunding")
        End Sub
        Public discontinuedProductException1 As DiscontinuedProductException = New Microsoft.Samples.Workflow.Compensation.DiscontinuedProductException
    End Class

    <SerializableAttribute()> _
     Public Class DiscontinuedProductException
        Inherits Exception
        Public Sub New()
            MyBase.New()
        End Sub

        Public Sub New(ByVal message As String)
            MyBase.New(message)
        End Sub

        Public Sub New(ByVal message As String, ByVal innerException As Exception)
            MyBase.New(message, innerException)
        End Sub

        Protected Sub New(ByVal info As SerializationInfo, ByVal context As StreamingContext)
            MyBase.New(info, context)
        End Sub

    End Class
End Namespace

File: sdk/samples/WFSamples/Technologies/Activities/Compensation/CS/Compensation/PurchaseOrder.cs
//---------------------------------------------------------------------
//  This file is part of the Windows Workflow Foundation SDK Code Samples.
// 
//  Copyright (C) Microsoft Corporation.  All rights reserved.
// 
//This source code is intended only as a supplement to Microsoft
//Development Tools and/or on-line documentation.  See these other
//materials for detailed information regarding Microsoft code samples.
// 
//THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
//KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
//IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
//PARTICULAR PURPOSE.
//---------------------------------------------------------------------

using System;
using System.ComponentModel;
using System.Runtime.Serialization;
using System.Workflow.Activities;
using System.Workflow.ComponentModel;


namespace Microsoft.Samples.Workflow.Compensation
{
    public sealed partial class PurchaseOrder : SequentialWorkflowActivity
    {
        public PurchaseOrder()
        {
            InitializeComponent();
        }

        void DiscontinuedProduct_ExecuteCode(object sender, EventArgs e)
        {
            Console.WriteLine("Product discontinued");
        }

        private void ReceiveOrderHandler(object sender, EventArgs e)
        {
            Console.WriteLine("Received order");
        }

        private void RefundHandler(object sender, EventArgs e)
        {
            Console.WriteLine("Refunding");
        }

        private void ProductCheckHandler(object sender, ConditionalEventArgs e)
        {
            e.Result = false;
        }

        private void ShipHandler(object sender, EventArgs e)
        {
            Console.WriteLine("Ship order");
        }

        private void WithdrawFundsHandler(object sender, EventArgs e)
        {
            Console.WriteLine("Withdraw funds");
        }

        public DiscontinuedProductException discontinuedProductException1 = new Microsoft.Samples.Workflow.Compensation.DiscontinuedProductException();

        
    }

    [SerializableAttribute()]
    public class DiscontinuedProductException : Exception
    {
        public DiscontinuedProductException()
            : base()
        {
        }

        public DiscontinuedProductException(string message)
            : base(message)
        {
        }

        public DiscontinuedProductException(string message, Exception innerException)
            : base(message, innerException)
        {
        }

        protected DiscontinuedProductException(SerializationInfo info, StreamingContext context)
            : base(info, context)
        {
        }
    }
}
